doc_src/cmds/string-join.rst

```diff
@@ -18,11 +18,26 @@ Description
 
 .. BEGIN DESCRIPTION
 
-``string join`` joins its *STRING* arguments into a single string separated by *SEP*, which can be an empty string. Exit status: 0 if at least one join was performed, or 1 otherwise. If ``-n`` or ``--no-empty`` is specified, empty strings are excluded from consideration (e.g. ``string join -n + a b "" c`` would expand to ``a+b+c`` not ``a+b++c``).
+Joins its *STRING* arguments into a single string separated by *SEP* (for ``string join``) or by the
+zero byte (NUL) (for ``string join0``).
+Exit status: 0 if at leas one join was performed, or 1 otherwise.
 
-``string join0`` joins its *STRING* arguments into a single string separated by the zero byte (NUL), and adds a trailing NUL. This is most useful in conjunction with tools that accept NUL-delimited input, such as ``sort -z``. Exit status: 0 if at least one join was performed, or 1 otherwise.
+``-n`` or ``--no-empty``: Exclude empty strings from consideration (e.g. ``string join -n + a b "" c`` would expand to ``a+b+c`` not ``a+b++c``).
 
-Because Unix uses NUL as the string terminator, passing the output of ``string join0`` as an *argument* to a command (via a :ref:`command substitution <expand-command-substitution>`) won't actually work. Fish will pass the correct bytes along, but the command won't be able to tell where the argument ends. This is a limitation of Unix' argument passing.
+``-q`` or ``--quiet``: Do not print the strings, only set the exit status as described above.
+
+**WARNING**: If you omit the ``--`` before the strings to be joined, any strings starting with
+``-`` will be treated as flag arguments, meaning they will most likely result in the command
+failing.
+This is also true if you specify a variable which expands to such a string instead of a literal string.
+If you don't need to append flag arguments at the end of the command,
+just always use ``--`` to avoid unwelcome surprises.
```

Comment:
this is a general problem but yeah, it's probably fine to include it here more prominently